src/v/redpanda/admin_server.h

```diff
@@ -146,6 +147,35 @@ class admin_server {
           });
     }
 
+    /**
+     * Variant of register_route for synchronous handlers.
```

Comment:
I think that synchronous should always be preferred when it can be made such. Do you have something more complicated on your mind about it?

Reply:
Ah I see. I was just surprised to see this since synchronous code can always be scheduled to run asynchronously, so it didn't seem like we were gaining anything here. But I agree, it's nice to be able to leverage the synchronous `path::set()` call, and I wonder if we avoid some extra reactor scheduling here.